crates/holochain/src/core/sys_validate.rs

```diff
@@ -605,7 +611,12 @@ fn make_store_element(element: Element) -> Option<(DhtOpHash, DhtOp)> {
 
     // Check the entry
     let maybe_entry_box = match element_entry {
-        ElementEntry::Present(e) => Some(e.into()),
+        ElementEntry::Present(e) => header
+            .entry_type()
+            .map_or(false, |et| {
+                matches!(et.visibility(), EntryVisibility::Public)
+            })
+            .then(|| e.into()),
         // This is ok because we weren't expecting an entry
         ElementEntry::NotApplicable | ElementEntry::Hidden => None,
```

Comment:
FYI `ElementEntry::Hidden` was intended to be for the case of private entries. Perhaps this can be fixed further upstream so that the private ops are `Hidden` at this point?

Reply:
Yeh but the entry isn't hidden to the author. That's the problem, it needs to be filtered out when being sent to the validation workflows/

Reply:
@freesig the thing I don't get is that `Element::new` doesn't even let you construct an Element with an `ElementEntry::Present` if the entry is private:

https://github.com/holochain/holochain/blob/414be16423878930604749166f5dcb854be30922/crates/holochain_zome_types/src/element.rs#L53-L72

So are we bypassing this constructor? It just seems like `Hidden` was supposed to prevent this kind of error from happening in the first place.

Reply:
That code doesn't prevent a private entry being used

Reply:
`(Some(entry), Some(_)) => ElementEntry::Present(entry), ` Ignores the visibility

Reply:
We want to be able to create elements with private entries if we are the author of the element

Reply:
Duh

Reply:
All this really does is explain why the entry might be missing

Reply:
OK I made a small PR with how I would prefer to see ElementEntry::Hidden used, yours is fine but if you like my change then you can merge it in here https://github.com/holochain/holochain/pull/1110